dask_sql/physical/rel/custom/predict.py

```diff
@@ -112,8 +110,9 @@ def convert(
             None,  # hints
         )
 
-        sql_outer_query = tmp_context._to_sql_string(outer_select)
-        df = tmp_context.sql(sql_outer_query)
```

Comment:
Seems reasonable to me but can we should a test to ensure we have not broken anything. Like are we sure that  `sql_outer_query ` does not add anything . 

Maybe we add a test to verify that the registered tables still line up and nothing temporary is left around.

Reply:
Good idea, I will add that in now